### PR TITLE
gh-codeql: init at 0-unstable-2024-06-09

### LINKED ISSUE #318587: Package request: gh-codeql
**Project description**

 GitHub CLI extension for working with CodeQL.

**Metadata**

* homepage URL: https://github.com/github/gh-codeql
* source URL: https://github.com/github/gh-codeql
* license: mit
* platforms: unix, linux, darwin

### PR DESCRIPTION
**Pull request by luftmensch-luftmensch**
## Description of changes
Closes #318587

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/gh/gh-codeql/package.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    owner = "github";
+    repo = "gh-codeql";
+    rev = "0c422da4b003d8de22e6f98a31888d7c0b080837";
+    hash = "sha256-0hpvUyFw3oC5UT6VU3rtMKR6RCKVvhJ6inY9gjOknrE=";
+  };
+
+  installPhase = ''
+    runHook preInstall
+
+    install -m755 -D $src/gh-codeql $out/bin/gh-codeql
```
Comment (eclairevoyant) [inline pkgs/by-name/gh/gh-codeql/package.nix:20]:
```suggestion
    install -m755 -D gh-codeql $out/bin/gh-codeql
```

**Review by eclairevoyant — COMMENTED** [review]
was this tested? it's downloading a zip here: https://github.com/github/gh-codeql/blob/main/gh-codeql#L160

**Comment by eclairevoyant** [comment]
Actually the binaries are in a different repo entirely: https://github.com/github/codeql-cli-binaries
And almost certainly need some patching/wrapping

**Comment by luftmensch-luftmensch** [comment]
Apparently I didn't test it properly and the provided URL in the issue didn't make me investigate enough

**Pull request closed without merging**